lib/errors/AuthApiError.js

```diff
@@ -10,19 +10,21 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-function AuthApiError(err, xhr) {
-  this.name = 'AuthApiError';
-  this.message = err.errorSummary;
-  this.errorSummary = err.errorSummary;
-  this.errorCode = err.errorCode;
-  this.errorLink = err.errorLink;
-  this.errorId = err.errorId;
-  this.errorCauses = err.errorCauses;
+export default class AuthApiError extends Error {
+  constructor(err, xhr = null) {
+    const message = err.errorSummary;
+    super(message); // 'Error' breaks prototype chain here
+    Object.setPrototypeOf(this, new.target.prototype); // restore prototype chain
```

Comment:
Not sure what's the issue with extending the Error class, but looks like there is a hack needed to make it work. Wondering if we can use any babel plugin to handle it or move the logic to a `BaseError` class to make future changes easier?

Reply:
A search for "typescript extend error prototype" will show more information on the issue, here is an example that also shows this solution: https://stackoverflow.com/questions/41102060/typescript-extending-error-class

A base `CustomError` class could possibly make this slightly simpler.

Reply:
I'll vote for a base error class, also I think it would be better to add the StackOverflow link in code.